diff --git a/scale.c b/scale.c
index 5857589..2f61594 100644
--- a/scale.c
+++ b/scale.c
@@ -423,7 +423,15 @@ static void scale_select_scaler(unsigned w, unsigned h, size_t pitch) {
 		return;
 	};
 
-	current_aspect_ratio = aspect_ratio > 0 ? aspect_ratio : ((double)w / (double)h);
+	double real_ratio = (double)w / (double)h;
+
+	/* Scaled mode: set correct aspect ratio for resolutions up to 320x224 */
+	if (real_ratio <= 10.0f / 7.0f) {
+		current_aspect_ratio = real_ratio;
+	} else {
+		/* 4:3 aspect ratio is forced for exotic resolutions (384x224 for CPS systems, 640x240 for PS1...) */
+		current_aspect_ratio = aspect_ratio > 0 ? aspect_ratio : real_ratio;
+	}
 
 	/* mame2000 sets resolutions / aspect ratio without notifying
 	 * of changes, new should always override old */
